src/apps/chahub/tasks.py

```diff
@@ -1,56 +1,165 @@
+    if not data:
+        logger.info(f'Nothing to send to Chahub at {endpoint}')
+        return
     try:
-        logger.info("Sending profile data to Chahub")
-        resp = send_to_chahub('profiles/', user_data_list, update=False)
+        logger.info(f"Sending all data to Chahub at {endpoint}")
+        resp = _send(endpoint=endpoint, data=data)
         logger.info(f"Response Status Code: {resp.status_code}")
-    # TODO: Will this catch timeouts and errors from our code? We should bubble up errors from send_to_chahub nicely.
+        if resp.status_code != 201:
+            logger.warning(f'ChaHub Response Content: {resp.content}')
     except ChahubException:
-        logger.info("There was a problem reaching Chahub, it is currently offline. Re-trying in 5 minutes.")
-        send_users_to_chahub.apply_async(eta=timezone.now() + timedelta(minutes=5))
+        logger.info("There was a problem reaching Chahub. Retry again later")
 
 
-@task(queue='site-worker')
-def do_chahub_retries(limit=None):
+def chahub_is_up():
     if not settings.CHAHUB_API_URL:
-        return
+        return False
 
     logger.info("Checking whether ChaHub is online before sending retries")
     try:
         response = requests.get(settings.CHAHUB_API_URL)
-        if response.status_code != 200:
-            return
+        if response.ok:
+            logger.info("ChaHub is online")
+            return True
+        else:
+            logger.info("Bad Status from ChaHub")
+            return False
     except requests.exceptions.RequestException:
         # This base exception works for HTTP errors, Connection errors, etc.
-        return
+        logger.info("Request Exception trying to access ChaHub")
+        return False
+
+
+def get_chahub_models():
+    from chahub.models import ChaHubSaveMixin
+    return ChaHubSaveMixin.__subclasses__()
+
 
-    logger.info("ChaHub is online, checking for objects needing to be re-sent to ChaHub")
-    chahub_models = ChaHubSaveMixin.__subclasses__()
+@app.task(queue='site-worker')
+def do_chahub_retries(limit=None):
+    if not chahub_is_up():
+        return
+    chahub_models = get_chahub_models()
+    logger.info(f'Retrying for ChaHub models: {chahub_models}')
     for model in chahub_models:
-        needs_retry = model.objects.filter(chahub_needs_retry=True)
+        batch_send_to_chahub(model, retry_only=True, limit=limit)
+        qs = model.objects.all_objects().filter(deleted=True)
+        if limit is not None:
+            qs = qs[:limit]
+        for obj in qs:
+            obj.delete()
 
-        if limit:
-            needs_retry = needs_retry[:limit]
 
-        for instance in needs_retry:
-            # Saving forces chahub update
-            instance.save()
+@app.task(queue='site-worker')
+def send_everything_to_chahub(limit=None):
+    if not chahub_is_up():
+        return
+    chahub_models = get_chahub_models()
+    for model in chahub_models:
```

Comment:
`for model in get_chahub_models():` instead of having the temp var `chahub_models`?

Reply:
👍